### PR TITLE
firewall: T4622: Add TCP MSS option

### PR DESCRIPTION
**Pull request by sever-sever**
## Change Summary

Ability to `drop|accept` packets based on TCP MSS size
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T4622

## Component(s) name

firewall
## How to test
VyOS configuration:
```
set firewall name FOO rule 10 action 'drop'
set firewall name FOO rule 10 protocol 'tcp'
set firewall name FOO rule 10 tcp flags syn
set firewall name FOO rule 10 tcp mss '1-500'
commit
```
nft rules:
```
vyos@r14# sudo nft -s list table ip filter
table ip filter {
...

	chain NAME_FOO {
		tcp flags & syn == syn tcp option maxseg size 1-500 counter drop comment "FOO-10"
		counter return comment "FOO default-action accept"
	}
}

```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Comment by c-po** [comment]
WOW I was not aware of that possibility

**Comment by c-po** [comment]
Very interesting feature

File: interface-definitions/include/firewall/tcp-flags.xml.i [hunk 1]
```diff
@@ -114,6 +114,23 @@
...
     </node>
+    <leafNode name="mss">
+      <properties>
+        <help>Maximum segment size (MSS)</help>
+        <valueHelp>
+          <format>u32:1-16384</format>
+          <description>Maximium segment size</description>
+        </valueHelp>
+        <valueHelp>
+          <format>&lt;min&gt;-&lt;max&gt;</format>
```
Comment (c-po) [inline interface-definitions/include/firewall/tcp-flags.xml.i:125]: Maybe we should have two `<valueHelp>` definitions. One for a single number, and another for the range.

File: smoketest/scripts/cli/test_firewall.py [hunk 2]
```diff
@@ -203,6 +203,10 @@ def test_basic_rules(self):
         self.cli_set(['firewall', 'name', 'smoketest', 'rule', '4', 'destination', 'port', '22'])
         self.cli_set(['firewall', 'name', 'smoketest', 'rule', '4', 'recent', 'count', '10'])
         self.cli_set(['firewall', 'name', 'smoketest', 'rule', '4', 'recent', 'time', 'minute'])
+        self.cli_set(['firewall', 'name', 'smoketest', 'rule', '5', 'action', 'accept'])
+        self.cli_set(['firewall', 'name', 'smoketest', 'rule', '5', 'protocol', 'tcp'])
+        self.cli_set(['firewall', 'name', 'smoketest', 'rule', '5', 'tcp', 'flags', 'syn'])
+        self.cli_set(['firewall', 'name', 'smoketest', 'rule', '5', 'tcp', 'mss', '501-1460'])
```
Comment (c-po) [inline smoketest/scripts/cli/test_firewall.py]: Please use a variable that defines "501-1460" and reference it here and also in the validation step below.

**Review by c-po — CHANGES_REQUESTED** [review]

File: interface-definitions/include/firewall/tcp-flags.xml.i (hunk 1 above)
Comment (sever-sever) [reply interface-definitions/include/firewall/tcp-flags.xml.i:125]: But there are 2 ValueHelp, aren’t?

Comment (sever-sever) [reply interface-definitions/include/firewall/tcp-flags.xml.i:125]: But there are 2 ValueHelp, aren’t?

Comment (c-po) [reply interface-definitions/include/firewall/tcp-flags.xml.i:125]: Sorty - I somehow missed that!

File: smoketest/scripts/cli/test_firewall.py (hunk 2 above)
Comment (sever-sever) [reply smoketest/scripts/cli/test_firewall.py]: Done!

**Review by c-po — APPROVED** [review]

**Comment by c-po** [comment]
@sarthurdev what you think?

**Review by sarthurdev — APPROVED** [review]
Looks good

**Pull request merged**